src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,27 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        executorService.submit(() -> listeners.forEach(listener -> listener.onEvent(event)));
```

Comment:
IMO it's better to dedicate a separate thread for each listener rather then one thread for all listeners, but this approach is also acceptable in some cases, so I won't reject because of it. However, think about the approach I described and decide what is better.